src/TimeZones.jl

```diff
@@ -43,6 +47,14 @@ function __init__()
     # Initialize the thread-local TimeZone cache (issue #342)
     _reset_tz_cache()
 
+    # Write out our compiled tzdata representations into a scratchspace
+    version = tzdata_version()
+    _COMPILED_DIR[] = _compiled_dir(version)
+
+    # Build the time zone data if required. Should only occur when `JULIA_TZ_VERSION` is
+    # specified we to a version not previously built.
+    isdir(_COMPILED_DIR[]) || build(version)
```

Comment:
Added this as otherwise a user who specifies `JULIA_TZ_VERSION` would be required to manually call build if they hadn't already built the tzdata version specified. As an additional benefit this should call build automatically in the cases where `deps/build.jl` is skipped. Possibly we could drop `deps/build.jl` entirely but that may not play nice with PackageCompiler.jl

Reply:
In https://github.com/JuliaTime/TimeZones.jl/issues/371 we may have seen an issue with PackageCompiler not playing nice with directory creation occurring during the package init. However, the use of scratch spaces may not have that problem and typically directories will be created during `deps/build.jl`

Reply:
I'm very much of two minds about this:
1. Having this causes the time zone data to be built if it's needed. For most users this means they shouldn't see the "please run `TimeZones.build()`" message which is much nicer in automated environments
2. In the case of a build failure the TimeZones.jl package won't allow you load the package and investigate the failure

I think this choice is important enough that I'll make it into a separate PR

Reply:
https://github.com/JuliaTime/TimeZones.jl/pull/390